score-xml/src/main/java/io/cloudslang/content/xml/utils/XmlUtils.java

```diff
@@ -109,4 +265,93 @@ private static String transformElementNode(Node node) throws TransformerExceptio
 
         return stringWriter.toString().trim();
     }
+
+    public static Document getDocument(CommonInputs commonInputs) throws Exception {
+        Document doc;
+        if (Constants.XML_PATH.equalsIgnoreCase(commonInputs.getXmlDocumentSource())) {
+            doc = XmlUtils.createDocumentFromFile(commonInputs.getXmlDocument(), commonInputs.getSecureProcessing());
+        } else {
+            doc = XmlUtils.parseXML(commonInputs.getXmlDocument(), commonInputs.getSecureProcessing());
+        }
+        return doc;
+    }
+
+    public static NamespaceContext getNamespaceContext(CommonInputs commonInputs, Document doc) throws Exception {
+        NamespaceContext context;
+        if (Constants.XML_PATH.equalsIgnoreCase(commonInputs.getXmlDocumentSource())) {
+            StringWriter writer = XmlUtils.getStringWriter(doc);
+            context = XmlUtils.getNamespaceContext(writer.toString(), "");
+        } else {
+            context = XmlUtils.getNamespaceContext(commonInputs.getXmlDocument(), "");
+        }
+        return context;
+    }
+
+    public static String createXmlDocumentFromUrl(CommonInputs commonInputs) throws ParserConfigurationException, SAXException, IOException {
+        if (commonInputs.getXmlDocument().startsWith("http")) {
```

Comment:
What if I want to provide a file path that starts with "http"? I would suggest checking first if it is indeed an URL and then if the protocol is http or https.